doc/pyoptsparse.bib

```diff
@@ -0,0 +1,559 @@
+@article{Halila2020d,
```

Comment:
Is the idea here to collect all publications that have used this particular package, or just MACH-Aero from the lab? Current the list is not exactly complete.

Reply:
The idea was to collect a non-exhaustive list of papers that have used pyOptSparse. I have opted to neglect many papers from the lab since most of them use this so the list would become too long, but it was not done in a systematic fashion---I basically picked some representative MACH papers with a variety of applications (BLI, transition, strut-braced wing, etc.). Do you think a more complete list would be beneficial?

Reply:
No, just curious. I am not sure if a complete listing here is beneficial, or feasible for that matter, as it will become outdated quickly since its currently manually updated. If we want that we could look into making a more automatically generated list by adding keywords to the original bib file. Not suggesting that for this PR though.  
If we want to highlight the breadth of works it select publication i.e., different frameworks or application, problem size etc., then we should just pick few representative papers (similar to you have). 
Also, I think we could include any Aerostructural work here as well under "MACH" or similar even though its not open-source, since those paper include some of the largest optimization that this group does with this package. Perhaps Prof. Martins will want to add specific papers here?
Its really a shame that we did not have (until recently) a separate paper what we could have cited specifically for pyOptSparse over pyOpt for older publication. It would have been easy to include a link to google scholar (or similar service) [for pyOpt](https://scholar.google.com/scholar?cites=14151378984480309081&as_sdt=2005&sciodt=0,5&hl=en) for a large amount of the publication. Maybe we should still do that and add a link to the [new one](https://scholar.google.com/scholar?cites=13781283857839029766&as_sdt=2005&sciodt=0,5&hl=en)?

Reply:
Yes exactly, and given how many papers from the lab uses pyOptSparse I'm inclined not to include every paper here. I had considered automated bib file process but the problem remains -- I don't want to just blanket include papers. See for example what OpenMDAO does on [their site](http://openmdao.org/publications/). I'll go through the list of papers here and clean up a little bit, and add some more aerostructural references. Other than that is there anything else you think I should do?

Nah that's okay I'm not really looking for a complete listing, but just some papers to highlight the wide-ranging applications.

Reply:
I think we should probably include some of the labs earlier more cited work like, like something from Gaetan, Graeme, or Peter work, e.g., "Multipoint high-fidelity aerostructural optimization of a transport aircraft configuration", and "Aerodynamic shape optimization investigations of the common research model wing benchmark" or something similar. Maybe Prof. Martins wants to highlight specific papers? We can also just update it later.

Reply:
Neither paper mentions pyOptSparse specifically although my understanding is they used the sparse interface. If you think that's still fine to cite then I will update this PR with that and call it a day I think.

Reply:
Yeah, they dont mention it explicitly, although I do think they probably did use it. Having said that we can just update this later.